src/VirtualClient/VirtualClient.Actions/CtsTraffic/CtsTrafficMetricsParser.cs

```diff
@@ -0,0 +1,121 @@
+
+        /// <inheritdoc/>
+        public override IList<Metric> Parse()
+        {
+            try
+            {
+                this.Preprocess();
+                this.Metrics = new List<Metric>();
+                MemoryStream mStrm = new MemoryStream(Encoding.UTF8.GetBytes(this.PreprocessedText));
+                using (var reader = new StreamReader(mStrm))
+                {
+                    var header = true;
+
+                    while (!reader.EndOfStream)
+                    {
+                        var line = reader.ReadLine();
+                        var values = line.Split(',');
+                        if (header)
+                        {
+                            header = false;
+                        }
+                        else
+                        {
+                            try
+                            {
+                                // bytes/sec that were sent within the TimeSlice period
+                                this.Metrics.Add(new Metric($"SendBps(TimeSlice-{values[0].Trim()})", Convert.ToDouble(values[1]), "B/s", MetricRelativity.Undefined));
+                            }
+                            catch
+                            {
+                                // do nothing as this result file has non-double values.
+                            }
+
+                            try
+                            {
+                                // bytes/sec that were received within the TimeSlice period
+                                this.Metrics.Add(new Metric($"RecvBps(TimeSlice-{values[0].Trim()})", Convert.ToDouble(values[2]), "B/s", MetricRelativity.Undefined));
+                            }
+                            catch
+                            {
+                                // do nothing as this result file has non-double values.
+                            }
+
+                            try
+                            {
+                                // count of established connections transmitting IO pattern data
+                                this.Metrics.Add(new Metric($"InFlight(TimeSlice-{values[0].Trim()})", Convert.ToDouble(values[3]), MetricRelativity.Undefined));
+                            }
+                            catch
+                            {
+                                // do nothing as this result file has non-double values.
+                            }
+
+                            try
+                            {
+                                // cumulative count of successfully completed IO patterns
+                                this.Metrics.Add(new Metric($"Completed(TimeSlice-{values[0].Trim()})", Convert.ToDouble(values[4]), MetricRelativity.HigherIsBetter));
+                            }
+                            catch
+                            {
+                                // do nothing as this result file has non-double values.
```

Comment:
If you may, could you wrap these try-metricsAdd-catch blocks into a method? Just to make this method less bulky